tests/helpers/testers.py

```diff
@@ -114,18 +114,24 @@ def _class_test(
     metric = pickle.loads(pickled_metric)
 
     for i in range(rank, NUM_BATCHES, worldsize):
-        batch_result = metric(preds[i], target[i])
+        b_kwargs_update = {k: v[i] for k, v in kwargs_update.items()}
```

Comment:
I have a open PR (https://github.com/PyTorchLightning/metrics/pull/90) where update can take a string. So I think we should only index tensors.
```suggestion
        b_kwargs_update = {k: v[i] if isinstance(v, Tensor) else v for k, v in kwargs_update.items()}
```

Reply:
update in https://github.com/PyTorchLightning/metrics/pull/137/commits/86e4015cb0bf85bd10c7f860ada0c8a0c93124d5